### PR TITLE
Use toolchain for build-logic build

### PR DESCRIPTION
**Pull request by blindpirate**

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test this

File: build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.build-logic.groovy-dsl-gradle-plugin.gradle.kts [hunk 1]
```diff
@@ -21,9 +24,12 @@ plugins {
     id("gradlebuild.ci-reporting")
 }
 
-java {
-    sourceCompatibility = JavaVersion.VERSION_11
-    targetCompatibility = JavaVersion.VERSION_11
+java.toolchain {
+    languageVersion.set(JavaLanguageVersion.of(11))
+    // Do not force Adoptium vendor for M1 Macs
+    if (System.getProperty("os.arch") != "aarch64") {
```
Comment (wolfs) [inline build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.build-logic.groovy-dsl-gradle-plugin.gradle.kts:30]: Can we extract this method somehow, so we don't need to repeat it so often?

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**